pkg/kv/kvserver/kvstorage/init.go

```diff
@@ -416,19 +415,6 @@ func loadReplicas(ctx context.Context, eng storage.Engine) ([]Replica, error) {
 
 	// INVARIANT: all replicas have a persisted full ReplicaID (i.e. a "ReplicaID from disk").
```

Comment:
Now that the invariant is true, it can be enforced early here, e.g. by the `s.set*` methods or the code that uses them. For example, on setting `s.setHardState` (and all other `set` methods except `setReplicaID`) we can check that the entry already exists. We would allow creating entries only while loading replica IDs.

This suggests loading in a different more convenient order:
1. Load all rangeID->replicaIDs. Now the set of range IDs is final, and for each rangeID there is a replicaID.
2. Load all rangeID -> descriptors, hard states, etc. These can assert that the touched range IDs already exist.

TODO?